nikola/plugin_categories.py

```diff
@@ -412,19 +415,19 @@ def split_metadata_from_text(self, source_text: str) -> (str, str):
                 # Necessary?
                 return split_result[0], split_result[-1]
 
-    def extract_text(self, source_text: str) -> 'typing.Dict[str, str]':
+    def extract_text(self, source_text: str) -> Dict[str, str]:
         """Split file, return metadata and the content."""
+        # TODO: The name and interface of this method is a mess and needs to be cleaned up.
         split = self.split_metadata_from_text(source_text)
-        if not split:
+        if len(split[0]) == 0:
```

Comment:
`not` is more Pythonic and can catch things like `None` if it somehow ended up there.

```suggestion
        if not split[0]:
```

Reply:
Done.